### PR TITLE
fix(pci.storages.databases): datatr-188 - distributed storage prices

### PR DESCRIPTION
**Pull request by jperchoc**
datatr-188

Signed-off-by: Jonathan Perchoc <jonathan.perchoc@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #DATATR-188
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Correctly compute database prices based on engine storage mode

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js [hunk 1]
```diff
@@ -346,9 +352,10 @@ export default class {
         : this.model.flavor.hourlyPricePerGB
       ).priceInUcents || 0;
     return (
-      this.model.plan.nodesCount *
-      (flavorPrice.priceInUcents +
-        additionalDiskPrice * this.model.disk.additionalDiskSize)
+      this.model.plan.nodesCount * flavorPrice.priceInUcents +
+      additionalDiskPrice *
+        this.model.disk.additionalDiskSize *
+        this.storageNodeFactor
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js:358]: **suggestion:**
```suggestion
      (this.model.plan.nodesCount * flavorPrice.priceInUcents) +
      (additionalDiskPrice *
        this.model.disk.additionalDiskSize *
        this.storageNodeFactor)
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js [hunk 2]
```diff
@@ -362,9 +369,10 @@ export default class {
         : this.model.flavor.hourlyPricePerGB
       ).tax || 0;
     return (
-      this.model.plan.nodesCount *
-      (flavorPrice.tax +
-        this.model.disk.additionalDiskSize * additionalDiskPrice)
+      this.model.plan.nodesCount * flavorPrice.tax +
+      this.model.disk.additionalDiskSize *
+        additionalDiskPrice *
+        this.storageNodeFactor
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js:375]: **suggestion:**
```suggestion
      (this.model.plan.nodesCount * flavorPrice.tax) +
      (this.model.disk.additionalDiskSize *
        additionalDiskPrice *
        this.storageNodeFactor)
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/add-node/add-node.routing.js [hunk 3]
```diff
@@ -13,8 +13,25 @@ export default /* @ngInject */ ($stateProvider) => {
...
+          // size of for one node, we need to divide by the number of nodes
+          if (dbEngine.isDistributedStorage) {
+            addedStorageByNode /= database.nodeNumber;
+          }
+          const additionalStoragePrice = Math.max(
+            0,
+            addedStorageByNode * currentFlavor.hourlyPricePerGB.priceInUcents,
+          );
+          return flavorPrice.priceInUcents + additionalStoragePrice;
+        },
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/add-node/add-node.routing.js:34]: **nitpick:** No need to block the state transition for this calculation. Better to perform this in ctrl.

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by ClaveauAlex — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js:358]: I have added those parenthesis, but they are automatically removed by the linter.

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js (hunk 2 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js:375]: Same as above

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/add-node/add-node.routing.js (hunk 3 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/add-node/add-node.routing.js]: Moved to the controller: https://github.com/ovh/manager/pull/8876/commits/081142295e998fee5b78dcc16e71b6c5b415e990

**Review by anooparveti — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**